flink-connectors/flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch6/Elasticsearch6UpsertTableSink.java

```diff
@@ -159,12 +165,24 @@ protected ElasticsearchUpsertTableSinkBase copy(
 		Optional.ofNullable(sinkOptions.get(BULK_FLUSH_BACKOFF_DELAY))
 			.ifPresent(v -> builder.setBulkFlushBackoffDelay(Long.valueOf(v)));
 
-		builder.setRestClientFactory(
-			new DefaultRestClientFactory(
+		if (Optional.ofNullable(sinkOptions.get(CONNECTOR_USERNAME)).isPresent() &&
+			Optional.ofNullable(sinkOptions.get(CONNECTOR_PASSWORD)).isPresent()) {
+			builder.setRestClientFactory(new AuthRestClientFactory(
+				sinkOptions.get(CONNECTOR_USERNAME),
+				sinkOptions.get(CONNECTOR_PASSWORD),
 				Optional.ofNullable(sinkOptions.get(REST_MAX_RETRY_TIMEOUT))
 					.map(Integer::valueOf)
 					.orElse(null),
-				sinkOptions.get(REST_PATH_PREFIX)));
+				sinkOptions.get(REST_PATH_PREFIX)
+			));
+		} else {
+			builder.setRestClientFactory(
+				new DefaultRestClientFactory(
+					Optional.ofNullable(sinkOptions.get(REST_MAX_RETRY_TIMEOUT))
+						.map(Integer::valueOf)
+						.orElse(null),
+					sinkOptions.get(REST_PATH_PREFIX)));
+		}
```

Comment:
Could you extract the common vairable before the if else condition? e.g. maxRetry, pathPrefix, username, password.